Week 3/Solution.java

```diff
@@ -0,0 +1,64 @@
+import java.io.*;
+import java.util.*;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+public class Solution {
+    public List<List<Integer>> threeSum(int[] n) {
+        List<List<Integer>> res = new  ArrayList<List<Integer>>();
+        if(n == null || n.length<3) return res;
+        int len = n.length;
+        HashSet<Integer> checked = new HashSet<Integer>();
+        for(int i = 0; i<len; i++){
+            int goal = n[i]*-1;
+            if(checked.add(goal)){
+                //TwoSum
+                HashSet<Integer> seen = new HashSet<Integer>();
+                for(int j = i+1;j<len;j++){
+                    if(seen.contains(goal-n[j])){
+                        addTriplet(res,n[i],n[j],(goal-n[j]));
+                    }else{
+                        seen.add(n[j]);
+                    }
+                } 
+            }
+        }
+        return res;
+    }
+    void addTriplet(List<List<Integer>> res, int one, int two, int three){
+        ArrayList<Integer> newTriplet = new ArrayList<Integer>();
+        newTriplet.add(one);
+        newTriplet.add(two);
+        newTriplet.add(three);
+        Collections.sort(newTriplet);
+        if(!res.contains(newTriplet)){
+            res.add(newTriplet);
+        }       
+    }
+    @Test
+   public void testThreeSum() {
```

Comment:
Solid testing. Good job picking equivalence classes and keeping your input small. The easier the examples are to step through, the better.